### DISCUSSION TITLE
can not override export class CreateLdChangeRoutingPointsOperationHandler extends GModelChangeRoutingPointsOperationHandler {

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-glsp/glsp#1284] [Q&A] [answered]
i want to save changed route point into my model state session.but when i write like this and register CreateLdChangeRoutingPointsOperationHandler into diagram-module ,an error occured
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/e49750f7-b690-42d8-9cdd-dab89b6fbb55)

**Comment by tortmayr** [comment]
@zrwsmd 
Please post the error message that you received otherwise we cannot help you.
At first glance your code looks correct.

**Reply by zrwsmd** [reply]
the internet was  not very good,here is error message
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/c0c3ae2e-e5d2-49a3-9587-04f04e5c26e9)

**Reply by zrwsmd** [reply]
my original edge route point is like below [{x:170,y:310},{x:170,y:209}]
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/69be37e8-fb39-4eca-8dba-f7a6c3b0f60b)
when i change route point and  debug GModelChangeRoutingPointsOperationHandler#executeChangeRoutingPoints() method ,
i found it has changed,
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/7e5718e4-848a-459d-b91e-70b9efb6c65f)

**Comment by zrwsmd** [comment, marked as answer]
so i  think may be i do not need to rewrite  this handler,i can get data from model index
